step/step.go

```diff
@@ -448,62 +451,196 @@ func (b XcodebuildBuilder) findTestBundle(opts findTestBundleOpts) (testBundle,
-	if exist, err := pathutil.IsPathExists(builtTestDir); err != nil {
-		return testBundle{}, fmt.Errorf("failed to check if built test directory exists (%s): %w", builtTestDir, err)
+	if exist, err := b.pathChecker.IsPathExists(builtTestDir); err != nil {
+		return "", fmt.Errorf("failed to check if built test directory exists (%s): %w", builtTestDir, err)
 	} else if !exist {
-		return testBundle{}, fmt.Errorf("built test directory does not exist at: %s", builtTestDir)
+		return "", fmt.Errorf("built test directory does not exist at: %s", builtTestDir)
 	}
-	b.logger.Printf("Built test directory: %s", builtTestDir)
 
-	return testBundle{
-		BuiltTestDir: builtTestDir,
-		XctestrunPth: xctestrunPth,
-		SYMRoot:      symRoot,
-	}, nil
+	return builtTestDir, nil
+}
+
+func xctestrunPathPattern(symRoot, scheme string) string {
+	return filepath.Join(symRoot, fmt.Sprintf("%s*.xctestrun", scheme))
+}
+
+func (b XcodebuildBuilder) exportXcodebuildLog(outputDir, xcodebuildLog string) error {
+	xcodebuildLogPath := filepath.Join(outputDir, xcodebuildLogBaseName)
+	if err := output.ExportOutputFileContent(xcodebuildLog, xcodebuildLogPath, xcodebuildLogPathEnvKey); err != nil {
+		return fmt.Errorf("failed to export %s, error: %w", xcodebuildLogPathEnvKey, err)
+	}
+	b.logger.Donef("The xcodebuild command log file path is available in %s env: %s", xcodebuildLogPathEnvKey, xcodebuildLogPath)
+	return nil
+}
+
+func (b XcodebuildBuilder) exportTestBundle(outputDir, builtTestDir string, xctestrunPths []string, defaultXctestrunPth string) error {
+	// BITRISE_TEST_BUNDLE_PATH
+	tmpDir, err := v2pathutil.NewPathProvider().CreateTempDir("test_bundle")
+	if err != nil {
+		return err
+	}
+
+	if err := command.CopyDir(builtTestDir, tmpDir, false); err != nil {
+		return err
+	}
+	copiedTestDirDestination := filepath.Join(tmpDir, filepath.Base(builtTestDir))
+
+	var copiedDefaultXctestrunPth string
+	var copiedXctestrunPths []string
+	for _, xctestrunPth := range xctestrunPths {
+		xctestrunDestination := filepath.Join(tmpDir, filepath.Base(xctestrunPth))
+		if err := command.CopyFile(xctestrunPth, xctestrunDestination); err != nil {
+			return err
+		}
+		copiedXctestrunPths = append(copiedXctestrunPths, xctestrunDestination)
+		if xctestrunPth == defaultXctestrunPth {
+			copiedDefaultXctestrunPth = xctestrunDestination
+		}
+	}
+
+	if err := tools.ExportEnvironmentWithEnvman(testBundlePathEnvKey, tmpDir); err != nil {
+		return err
+	}
+	b.logger.Donef("The test bundle directory is available in %s env: %s", testBundlePathEnvKey, tmpDir)
+
+	// BITRISE_TEST_BUNDLE_ZIP_PATH
+	testBundleZipPth := filepath.Join(outputDir, "testbundle.zip")
+	factory := v2command.NewFactory(env.NewRepository())
+	args := []string{"-r", testBundleZipPth, filepath.Base(copiedTestDirDestination)}
+	for _, xctestrunPth := range copiedXctestrunPths {
+		args = append(args, filepath.Base(xctestrunPth))
+	}
+	zipCmd := factory.Create("zip", args, &v2command.Opts{
```

Comment:
Does not this recreate the full file path inside the zip file? I tried it locally on my Mac and when I did 
```
zip -r test.zip /path/to/file1 /path/to/file2 /path/to/file3
```
where all of the files were in the same directory created a zip file with folders to /path/to instead of just the files. I could only have the files without the extra directory structure if I first went to the directory I wanted to compress and ran
```
zip -r test.zip .
```
Could you check a created `testbundle.zip`? And if the same is happening there then is this our expected behaviour?

Reply:
testbundle.zip works well, the reason is that the zip command's workdir is the test bundle dir and all the content is added by filenames.

```
	args := []string{"-r", testBundleZipPth, filepath.Base(copiedTestDirDestination)}
	for _, xctestrunPth := range copiedXctestrunPths {
		args = append(args, filepath.Base(xctestrunPth))
	}
	zipCmd := factory.Create("zip", args, &v2command.Opts{
		Dir: tmpDir,
	})
```